feat: add role and ARIA attributes to master-detail-layout

web-padawan opened the pull request:
## Description

Fixes #8814

## Type of change

- Feature

web-padawan commented:
Note: I eventually decided to create a separate suite for ARIA tests as well, but IMO this still can be merged.

vursen commented on packages/master-detail-layout/test/aria.test.js:
```diff
@@ -0,0 +1,56 @@
+    detail = layout.shadowRoot.querySelector('[part="detail"]');
+  });
+
+  it('should set role to dialog on the detail part in the overlay mode', () => {
```
**idea:** These tests look like good candidates for being snapshot tests.

web-padawan replied:
Yes, I was not quite sure about this. Modified tests to also cover clearing `overlay` and `stack` mode for now.

web-padawan commented:
Updated the PR based on the internal discussion to not use `role="region"` since that should be up to the user.
Also changed to not expose `overlay` and `stack` as public properties for now (we can reconsider if if needed).
